### PR TITLE
Documentation update

### PR DESCRIPTION
**Pull request by DIOHz0r**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | N/A

*Please update this template with something that matches your PR*

Hello, this is a minor update for a lot of documentation blocks

### CODE REVIEW

File: inc/autoload.function.php [hunk 1]
```diff
@@ -301,9 +301,9 @@ function _nx($ctx, $sing, $plural, $nb, $domain = 'glpi') {
 /**
  * Classes loader
  *
- * @param string $classname : class to load
+ * @param string $classname to load
```
Comment (trasher) [inline inc/autoload.function.php]: Do not assume variable name is part of the comment

File: inc/commondbtm.class.php [hunk 2]
```diff
@@ -134,12 +134,15 @@ static function getTable($classname = null) {
...
     *
     * @return void
-   **/
+    */
    static function forceTable($table) {
       $_SESSION['glpi_table_of'][get_called_class()] = $table;
    }
 
 
+   /**
```
Comment (trasher) [inline inc/commondbtm.class.php]: This block is not complete.

File: inc/commondbtm.class.php [hunk 3]
```diff
@@ -456,7 +459,7 @@ function updateInDB($updates, $oldvalues = []) {
    /**
     * Add an item to the database
     *
-    * @return integer|boolean new ID of the item is insert successfull else false
+    * @return boolean new ID of the item is insert successful else false
```
Comment (trasher) [inline inc/commondbtm.class.php]: integer|boolean was correct

File: inc/autoload.function.php (hunk 1 above)
Comment (trasher) [reply inc/autoload.function.php]: This comment is accurate for several changes here.

File: inc/commondbtm.class.php (hunk 3 above)
Comment (trasher) [reply inc/commondbtm.class.php]: Globally, be very carefull changing return types

**Comment by trasher** [comment]
Made a few comments. This one must be rebased on master (9.3/bf branch would be acceptable if you really want). Ping us once rebased so we change PR targets. Ping us also if you need help on rebasing (do not open another PR for the same change).

**Comment by DIOHz0r** [comment]
@trasher I have resabased the PR on master branch, could you please change the PR target?

**Comment by DIOHz0r** [comment]
Hello, This PR has a lot of time waiting for review or be merge. Do you have any update for this?

File: inc/autoload.function.php [hunk 4]
```diff
@@ -70,7 +70,7 @@ function isAPI() {
  *
  * @param string $classname class name to analyze
  *
- * @return boolean[object false or an object containing plugin name and class name
+ * @return array|boolean
```
Comment (cedric-anne) [inline inc/autoload.function.php]: In this case, structure of returned array should be documented.

File: inc/autoload.function.php [hunk 5]
```diff
@@ -252,9 +252,9 @@ function _nx($ctx, $sing, $plural, $nb, $domain = 'glpi') {
 /**
  * Classes loader
  *
- * @param string $classname : class to load
+ * @param string $classname class to load
  *
- * @return void
+ * @return boolean
```
Comment (cedric-anne) [inline inc/autoload.function.php]: It returns `void|boolean`.

File: inc/commondbtm.class.php [hunk 6]
```diff
@@ -132,12 +132,17 @@ static function getTable($classname = null) {
...
+    */
    static function forceTable($table) {
       $_SESSION['glpi_table_of'][get_called_class()] = $table;
    }
 
 
+   /**
+    * Retrieve the foreign key of a table related to the called class
+    *
+    * @return mixed
```
Comment (cedric-anne) [inline inc/commondbtm.class.php]: This function returns a string as it returns result of `getForeignKeyFieldForTable()` function that is stored in cache variable.

File: inc/commondbtm.class.php [hunk 7]
```diff
@@ -1287,8 +1293,8 @@ function update(array $input, $history = 1, $options = []) {
    /**
     * Forward entity information to linked items
     *
-    * @return void
-   **/
+    * @return boolean
```
Comment (cedric-anne) [inline inc/commondbtm.class.php]: It returns `void|boolean`.

File: inc/commondbtm.class.php [hunk 8]
```diff
@@ -2313,8 +2319,8 @@ function showFormButtons($options = []) {
     * @param array   $options Array of possible options:
     *     - withtemplate : 1 for newtemplate, 2 for newobject from template
     *
-    * @return integer|void value of withtemplate option (exit of no right)
-   **/
+    * @return mixed|string
```
Comment (cedric-anne) [inline inc/commondbtm.class.php]: It returns the value of option `withtemplate` or empty string. So documentation should tell the returned value is `integer|string` and that it is corresponding to `withtemplate` option value.

**Comment by DIOHz0r** [comment]
Hi, I have updated again the PR with your comments

File: inc/html.class.php [hunk 9]
```diff
@@ -959,15 +949,12 @@ static function getConfirmationOnActionScript($string, $additionalactions = '')
...
-    * @param $id                 HTML ID of the progress bar
-    * @param $options    array   progress status
-    *                    - create    do we have to create it ?
-    *                    - message   add or change the message
-    *                    - percent   current level
-    *
-    *
-    * @return nothing (display)
-    **/
+    * @param integer $id        HTML ID of the progress bar
```
Comment (cedric-anne) [inline inc/html.class.php]: It is an HTML ID, so it is a string.

File: inc/html.class.php [hunk 10]
```diff
@@ -1373,7 +1350,8 @@ static function getMenuInfos() {
     * @since  9.2
     *
     * @param  boolean $force do we need to force regeneration of $_SESSION['glpimenu']
-    * @return array          the menu array
+    *
+    * @return array|string
```
Comment (cedric-anne) [inline inc/html.class.php]: It always return an array.

File: inc/html.class.php [hunk 11]
```diff
@@ -2312,12 +2289,12 @@ static function getMassiveActionCheckBox($itemtype, $id, array $options = []) {
...
-    * @param $itemtype             Massive action itemtype
-    * @param $id                   ID of the item
-    * @param $options      array
+    * @param string  $itemtype Massive action itemtype
+    * @param integer $id       ID of the item
+    * @param array   $options
     *
-    * @return show checkbox
-   **/
+    * @return string
```
Comment (cedric-anne) [inline inc/html.class.php]: It returns void.

File: inc/html.class.php [hunk 12]
```diff
@@ -660,13 +653,11 @@ static function displayAjaxMessageAfterRedirect() {
...
-    * @param $ref_title       Title to display (default '')
-    * @param $ref_btts        Extra items to display array(link=>text...) (default '')
-    *
-    * @return nothing
-   **/
+    * @param string $ref_pic_link   Path to the image to display (default '')
+    * @param string $ref_pic_text   Alt text of the icon (default '')
+    * @param string $ref_title      Title to display (default '')
+    * @param string $ref_btts       Extra items to display array(link=>text...) (default '')
+    */
```
Comment (cedric-anne) [inline inc/html.class.php]: It returns void.

File: inc/html.class.php [hunk 13]
```diff
@@ -959,15 +949,12 @@ static function getConfirmationOnActionScript($string, $additionalactions = '')
...
-    *
-    *
-    * @return nothing (display)
-    **/
+    * @param integer $id        HTML ID of the progress bar
+    * @param array   $options   progress status
+    *                           - create    do we have to create it ?
+    *                           - message   add or change the message
+    *                           - percent   current level
+    */
```
Comment (cedric-anne) [inline inc/html.class.php]: It returns void.

File: inc/html.class.php [hunk 14]
```diff
@@ -1010,10 +997,8 @@ static function progressBar($id, array $options = []) {
    /**
     * Create a Dynamic Progress Bar
     *
-    * @param $msg initial message (under the bar) (default '&nbsp;')
-    *
-    * @return nothing
-    **/
+    * @param string $msg initial message (under the bar) (default '&nbsp;')
+    */
```
Comment (cedric-anne) [inline inc/html.class.php]: It returns void.

File: inc/html.class.php [hunk 15]
```diff
@@ -1027,10 +1012,8 @@ static function createProgressBar($msg = "&nbsp;") {
    /**
     * Change the Message under the Progress Bar
     *
-    * @param $msg message under the bar (default '&nbsp;')
-    *
-    * @return nothing
-   **/
+    * @param string $msg message under the bar (default '&nbsp;')
+    */
```
Comment (cedric-anne) [inline inc/html.class.php]: It returns void.

File: inc/html.class.php [hunk 16]
```diff
@@ -1041,12 +1024,10 @@ static function changeProgressBarMessage($msg = "&nbsp;") {
...
-    * @param $crt   Current Value (less then $max)
-    * @param $tot   Maximum Value
-    * @param $msg   message inside the bar (default is %) (default '')
-    *
-    * @return nothing
-   **/
+    * @param integer $crt   Current Value (less then $max)
+    * @param integer $tot   Maximum Value
+    * @param string  $msg   message inside the bar (default is %) (default '')
+    */
```
Comment (cedric-anne) [inline inc/html.class.php]: It returns void.

File: inc/html.class.php [hunk 17]
```diff
@@ -1071,15 +1052,13 @@ static function changeProgressBarPosition($crt, $tot, $msg = "") {
...
-    *
-    * @return nothing
-   **/
+    * @param integer $width      width of the progress bar
+    * @param integer $percent    percent of the progress bar
+    * @param array   $options    possible options:
+    *                            - title : string title to display (default Progesssion)
+    *                            - simple : display a simple progress bar (no title / only percent)
+    *                            - forcepadding : boolean force str_pad to force refresh (default true)
+    */
```
Comment (cedric-anne) [inline inc/html.class.php]: It returns void.

File: inc/html.class.php [hunk 18]
```diff
@@ -1126,13 +1105,11 @@ static function displayProgressBar($width, $percent, $options = []) {
...
-    * @param $item      item corresponding to the page displayed (default 'none')
-    * @param $option    option corresponding to the page displayed (default '')
-    *
-    * @return nothing
-   **/
+    * @param string $title    title used for the page (default '')
+    * @param string $sector   sector in which the page displayed is (default 'none')
+    * @param string $item     item corresponding to the page displayed (default 'none')
+    * @param string $option   option corresponding to the page displayed (default '')
+    */
```
Comment (cedric-anne) [inline inc/html.class.php]: It returns void.

File: inc/html.class.php [hunk 19]
```diff
@@ -2275,9 +2254,7 @@ static function getCheckbox(array $options) {
     * @since 0.85
     *
     * @param $options   array
-    *
-    * @return nothing (display only)
-   **/
+    */
```
Comment (cedric-anne) [inline inc/html.class.php]: It returns void.

File: inc/html.class.php [hunk 20]
```diff
@@ -2328,10 +2305,8 @@ static function showMassiveActionCheckBox($itemtype, $id, array $options = []) {
     *
     * @since 0.84
     *
-    * @param $name given name/id to the form   (default '')
-    *
-    * @return nothing / display item
-   **/
+    * @param string $name given name/id to the form   (default '')
+    */
```
Comment (cedric-anne) [inline inc/html.class.php]: It returns void.

File: inc/html.class.php [hunk 21]
```diff
@@ -3048,18 +3025,19 @@ static function getGenericDateTimeSearchItems($options) {
...
+    * @param string  $val            date / datetime value passed
+    * @param boolean $force_day      force computation in days (false by default)
+    * @param string  $specifictime   set specific timestamp (default '')
     *
-    * @return computed date / datetime value
-    * @see showGenericDateTimeSearch
-   **/
+    * @see   showGenericDateTimeSearch
+    *
+    * @return false|string
```
Comment (cedric-anne) [inline inc/html.class.php]: Should only return string. Returning false is a bug and should be handled by function itself.

File: inc/html.class.php [hunk 22]
```diff
@@ -3163,18 +3141,20 @@ static function computeGenericDateTimeSearch($val, $force_day = false, $specific
...
+    *                   (false) the html
+    *                   - add_now, boolean to precise if we need to add to dates array, an entry
+    *                   for now time (with now class)
+    *
+    * @see   showGenericDateTimeSearch
     *
-    * @return array of posible values
-    * @see showGenericDateTimeSearch
-   **/
+    * @return string
```
Comment (cedric-anne) [inline inc/html.class.php]: It can also returns void if echoed  by function itself.

File: inc/html.class.php [hunk 23]
```diff
@@ -3269,22 +3249,21 @@ static function showProfileSelecter($target) {
...
+    *                        - contentid : string / id for the content html container (default auto generated) (used for ajax)
+    *                        - link : string / link to put on displayed image if contentid is empty
+    *                        - linkid : string / html id to put to the link link (used for ajax)
+    *                        - linktarget : string / target for the link
+    *                        - popup : string / popup action : link not needed to use it
+    *                        - img : string / url of a specific img to use
+    *                        - display : boolean / display the item : false return the datas
+    *                        - autoclose : boolean / autoclose the item : default true (false permit to scroll)
+    *
+    * @return string
```
Comment (cedric-anne) [inline inc/html.class.php]: It can also returns void if echoed by function itself.

File: inc/html.class.php [hunk 24]
```diff
@@ -3404,8 +3383,8 @@ static function showToolTip($content, $options = []) {
     *    - rand    : integer / pre-exsting random value
     *    - attrs   : array of attributes to add (['name' => 'value']
     *
-    * @return void|string
-   **/
+    * @return string
```
Comment (cedric-anne) [inline inc/html.class.php]: It can also returns void if echoed by function itself.

File: inc/html.class.php [hunk 25]
```diff
@@ -3485,13 +3464,13 @@ class='autocompletion-text-field'";
...
-    * @param $display    boolean display or get js script (true by default)
-    * @param $readonly   boolean editor will be readonly or not
+    * @param string  $name     the html textarea name to use
+    * @param string  $rand     the html textarea to use (if empty no image paste system)(default '')
+    * @param boolean $display  choose if display or get js script (true by default)
+    * @param boolean $readonly editor will be readonly or not
     *
-    * @return nothing
-   **/
+    * @return string
```
Comment (cedric-anne) [inline inc/html.class.php]: It can also returns void if echoed by function itself.

File: inc/html.class.php [hunk 26]
```diff
@@ -3598,10 +3577,10 @@ static function initEditorSystem($name, $rand = '', $display = true, $readonly =
     *
     * @since 9.2
     *
-    * @param $content : content to convert in html
+    * @param mixed $content to convert in html
```
Comment (cedric-anne) [inline inc/html.class.php]: Usage is made for a string. So it should be documented as well.

File: inc/html.class.php [hunk 27]
```diff
@@ -3598,10 +3577,10 @@ static function initEditorSystem($name, $rand = '', $display = true, $readonly =
...
     *
-    * @return $content
-   **/
+    * @return mixed
```
Comment (cedric-anne) [inline inc/html.class.php]: It has been made to be called using a string and return a string.

File: inc/html.class.php [hunk 28]
```diff
@@ -3655,7 +3633,7 @@ static function setRichTextContent($name, $content, $rand, $readonly = false) {
     *
     * @return void|string
    **/
-   static function printAjaxPager($title, $start, $numrows, $additional_info = '', $display = true, $additional_params = '') {
+   static function printAjaxPager($title, $start, $numrows, $additional_info = '', $display = true) {
```
Comment (cedric-anne) [inline inc/html.class.php]: `$additional_params` should not be removed. Maybe a problem when rebasing.

File: inc/html.class.php [hunk 29]
```diff
@@ -3746,12 +3724,10 @@ static function printAjaxPager($title, $start, $numrows, $additional_info = '',
...
-    * @param $tab          the array to display
-    * @param $pad          Pad used (default 0)
-    * @param $jsexpand     Expand using JS ? (default  false)
-    *
-    * @return nothing
-   **/
+    * @param array   $tab      the array to display
+    * @param integer $pad      pad used (default 0)
+    * @param boolean $jsexpand expand using JS ? (default  false)
+    */
```
Comment (cedric-anne) [inline inc/html.class.php]: It returns void.

File: inc/html.class.php [hunk 30]
```diff
@@ -3802,22 +3778,20 @@ static function printCleanArray($tab, $pad = 0, $jsexpand = false) {
...
+    * @param string  $target                 page would be open when click on the option
+    *                                        (last,previous etc)
+    * @param string  $parameters             parameters would be passed on the URL.
+    * @param integer $item_type_output       item type display - if >0 display export PDF et Sylk
+    *                                        form (default 0)
+    * @param integer $item_type_output_param item type parameter for export (default 0)
+    * @param string  $additional_info        Additional information to display (default '')
     *
-   **/
+    */
```
Comment (cedric-anne) [inline inc/html.class.php]: It returns void.

File: inc/html.class.php [hunk 31]
```diff
@@ -4008,18 +3982,20 @@ static function makeTitle ($string, $num, $tot) {
...
+    * @param array        $fields     field name => field  value
+    * @param string       $btimage    button image uri (optional)   (default '')
+    *                                 If image name starts with "fa-", il will be turned into a
+    *                                 font awesome element rather than an image.
+    * @param string       $btoption   optional button option        (default '')
+    * @param string       $confirm    optional confirm message      (default '')
     *
-    * @since 0.84
-   **/
+    * @since version 0.84
```
Comment (cedric-anne) [inline inc/html.class.php]: Remove `version`.

File: inc/html.class.php [hunk 32]
```diff
@@ -4084,16 +4060,16 @@ static function getSimpleForm($action, $btname, $btlabel, Array $fields = [], $b
...
+    * @param string|array $btname     button name
+    * @param string       $btlabel    button label
+    * @param array        $fields     field name => field  value
+    * @param string       $btimage    button image uri (optional) (default '')
+    * @param string       $btoption   optional button option (default '')
+    * @param string       $confirm    optional confirm message (default '')
     *
-    * @since 0.83.3
-   **/
+    * @since version 0.83.3
```
Comment (cedric-anne) [inline inc/html.class.php]: Remove `version`.

File: inc/html.class.php [hunk 33]
```diff
@@ -4708,11 +4685,11 @@ static function parseAttributes($options = []) {
     *
     * @since 0.85
     *
-    * @param $key       The name of the attribute to create
-    * @param $value     The value of the attribute to create.
+    * @param string $key       name of the attribute to create
+    * @param mixed  $value     value of the attribute to create.
```
Comment (cedric-anne) [inline inc/html.class.php]: `string|array`

File: inc/html.class.php [hunk 34]
```diff
@@ -5317,8 +5295,8 @@ static function showCheckboxMatrix(array $columns, array $rows, array $options =
    /**
     * This function provides a mecanism to send html form by ajax
     *
-    * @since 9.1
-   **/
+    * @since version 9.1
```
Comment (cedric-anne) [inline inc/html.class.php]: Remove version.

File: inc/html.class.php [hunk 35]
```diff
@@ -5355,8 +5333,8 @@ static function ajaxForm($selector, $success = "console.log(html);") {
     * In this function, we redefine 'window.alert' javascript function
     * by a jquery-ui dialog equivalent (but prettier).
     *
-    * @since 9.1
-   **/
+    * @since version 9.1
```
Comment (cedric-anne) [inline inc/html.class.php]: Remove version.

File: inc/toolbox.class.php [hunk 36]
```diff
@@ -169,8 +168,8 @@ static function substr($str, $start, $length = -1) {
     *
     * @param $str string
     *
-    * @return lower case string
-   **/
+    * @return mixed|string
```
Comment (cedric-anne) [inline inc/toolbox.class.php]: Documentation says it will always return a string.

File: inc/toolbox.class.php [hunk 37]
```diff
@@ -181,8 +180,8 @@ static function strtolower($str) {
     *
     * @param $str string
     *
-    * @return upper case string
-   **/
+    * @return mixed|string
```
Comment (cedric-anne) [inline inc/toolbox.class.php]: Documentation says it will always return a string.

File: inc/toolbox.class.php [hunk 38]
```diff
@@ -282,10 +281,10 @@ static function decrypt($string, $key) {
     *
     * @param $value array or string: item to prevent (array or string)
     *
-    * @return clean item
+    * @return array|mixed|null|resource
```
Comment (cedric-anne) [inline inc/toolbox.class.php]: `array|string|resource|null`

File: inc/toolbox.class.php [hunk 39]
```diff
@@ -306,10 +305,10 @@ static function clean_cross_side_scripting_deep($value) {
     *
     * @param $value  array or string   item to unclean from clean_cross_side_scripting_deep
     *
-    * @return unclean item
+    * @return array|mixed|null|resource
```
Comment (cedric-anne) [inline inc/toolbox.class.php]: `array|string|resource|null`

File: inc/toolbox.class.php [hunk 40]
```diff
@@ -332,10 +331,10 @@ static function unclean_cross_side_scripting_deep($value) {
     *
     * @param $value array or string: item to unclean from clean_cross_side_scripting_deep
     *
-    * @return unclean item
+    * @return array|mixed|null|resource|string
```
Comment (cedric-anne) [inline inc/toolbox.class.php]: `array|string|resource|null`

File: inc/toolbox.class.php [hunk 41]
```diff
@@ -692,8 +697,8 @@ static function userErrorHandlerDebug($errno, $errmsg, $filename, $linenum) {
     * @param $debug_vars   Boolean  (default NULL)
     * @param $log_in_files Boolean  (default NULL)
     *
-    * @since 0.84
-   **/
+    * @since version 0.84
```
Comment (cedric-anne) [inline inc/toolbox.class.php]: Remove version.

File: inc/toolbox.class.php [hunk 42]
```diff
@@ -923,14 +925,16 @@ static function getMemoryLimit($ininame = 'memory_limit') {
...
    /**
     * Check is current memory_limit is enough for GLPI
+    * returned values:
+    *    0 if PHP not compiled with memory_limit support
+    *    1 no memory limit (memory_limit = -1)
+    *    2 insufficient memory for GLPI
+    *    3 enough memory for GLPI
     *
-    * @since 0.83
+    * @since version 0.83
```
Comment (cedric-anne) [inline inc/toolbox.class.php]: Remove version.

File: inc/toolbox.class.php [hunk 43]
```diff
@@ -1105,12 +1114,15 @@ static function commonCheckForUseGLPI($isInstall = false) {
...
    /**
     * Check SELinux configuration
+    * returned values meaning:
+    *    0: OK
+    *    1: Warning
+    *    2: Error
     *
-    * @since 0.84
-    * @param $fordebug    Boolean true is displayed in system information
+    * @since version 0.84
```
Comment (cedric-anne) [inline inc/toolbox.class.php]: Remove version.

File: inc/toolbox.class.php [hunk 44]
```diff
@@ -1105,12 +1114,15 @@ static function commonCheckForUseGLPI($isInstall = false) {
...
     *
```
Comment (cedric-anne) [inline inc/toolbox.class.php]: Seems like you removed documentation of `$fordebug` param.

File: inc/toolbox.class.php [hunk 45]
```diff
@@ -2500,16 +2520,16 @@ static function checkValidReferer() {
...
     * @param $file   string      path of the file
-    * @param $type   string      check if $file is the correct type (false by default)
+    * @param boolean|string $type string      check if $file is the correct type (false by default)
+    * @return string|boolean
     *
-    * @return string (if $type not given) else boolean
-    *
-   **/
+    * @return string|boolean
```
Comment (cedric-anne) [inline inc/toolbox.class.php]: Return tag is present twice.

File: inc/toolbox.class.php [hunk 46]
```diff
@@ -2723,11 +2743,11 @@ static function convertTagToImage($content_text, CommonDBTM $item, $doc_data = [
...
     * @since 9.2
     *
-    * @param $content_html   html content of input
-    * @param $force_update   force update of content in item (false by default)
+    * @param string    $content_html    html content of input
+    * @param boolean   $force_update    force update of content in item (false by default)
     *
-    * @return html content
-   **/
+    * @return mixed|string
```
Comment (cedric-anne) [inline inc/toolbox.class.php]: It will always return string if correctly used passing a strig for first parameter.

File: inc/toolbox.class.php [hunk 47]
```diff
@@ -2750,10 +2770,10 @@ static function convertImageToTag($content_html, $force_update = false) {
     * @since 9.2
     *
     * @param string $content   html content of input
-    * @param array $tags       list of tags to clen
+    * @param array $tags       list of tags to clean
     *
-    * @return html content
-   **/
+    * @return array|mixed|string
```
Comment (cedric-anne) [inline inc/toolbox.class.php]: It will always return string if correctly used passing a strig for first parameter.

**Comment by cedric-anne** [comment]
Hi,

I added a lot of comments as I did not read all your PR yet.

It takes me 2 hours to read all your changes, and taking so much time to read a PR will not always be possible. Please, next time, make a maximum of 10 documentation blocks updates for a PR, in order to make validation easier for us. You can create multiple PR if you want.

Regards

File: inc/html.class.php [hunk 48]
```diff
@@ -129,11 +130,11 @@ static function entities_deep($value) {
    /**
     * Convert a date YY-MM-DD to DD-MM-YY for calendar
     *
-    * @param $time       date  date to convert
-    * @param $format           (default null)
+    * @param mixed $time datetime to convert
```
[the remark this answers is not in the document]
Comment (DIOHz0r) [reply inc/html.class.php]: Which argument do you refer?. If you are talking about $time, on the first if your are checking for different values so you certainly don't know what are you going to receive, that's why the mixed type.

File: inc/html.class.php [hunk 49]
```diff
@@ -178,11 +179,11 @@ static function convDate($time, $format = null) {
    /**
     * Convert a date YY-MM-DD HH:MM to DD-MM-YY HH:MM for display in a html table
     *
-    * @param $time        datetime  datetime to convert
-    *  @param $format               (default null)
+    * @param mixed $time datetime to convert
```
[the remark this answers is not in the document]
Comment (DIOHz0r) [reply inc/html.class.php]: same as above

File: inc/html.class.php [hunk 50]
```diff
@@ -210,8 +211,8 @@ static function cleanInputText($string) {
     *
     * @param $url string URL
     *
-    * @return clean URL
-   **/
+    * @return mixed
```
[the remark this answers is not in the document]
Comment (DIOHz0r) [reply inc/html.class.php]: Same comment about preg_replace here

File: inc/html.class.php [hunk 51]
```diff
@@ -274,8 +275,8 @@ static function resume_name($string, $length = 255) {
     *
     * @param $value string: string value
     *
-    * @return clean value
-   **/
+    * @return array|mixed
```
[the remark this answers is not in the document]
Comment (DIOHz0r) [reply inc/html.class.php]: str_replace function returns mixed values too
http://php.net/manual/en/function.str-replace.php#refsect1-function.str-replace-returnvalues

File: inc/html.class.php [hunk 52]
```diff
@@ -298,13 +299,13 @@ static function cleanPostForTextArea($value) {
...
-    *                               (false by default)
-    * @param $forcedecimal integer  Force decimal number (do not use default value) (default -1)
+    * @param float   $number       Number to display
+    * @param boolean $edit         display number for edition ? (id edit use . in all  case) false
+    *                              by default)
+    * @param integer $forcedecimal Force decimal number (do not use default value) (default -1)
     *
-    * @return formatted number
-   **/
+    * @return mixed|string
```
[the remark this answers is not in the document]
Comment (DIOHz0r) [reply inc/html.class.php]: You could return mixing values with str_replace

File: inc/html.class.php (hunk 12 above)
Comment (DIOHz0r) [reply inc/html.class.php]: echo getMassiveActionCheckBox returns a string and so it does getCheckbox which is called by getMassiveActionCheckBox, so at the end this function is returning a string

File: inc/html.class.php (hunk 13 above)
Comment (DIOHz0r) [reply inc/html.class.php]: I only fix the arguments here but actually in some cases this prints html so you are returning a string, does it count?

File: inc/html.class.php (hunk 14 above)
Comment (DIOHz0r) [reply inc/html.class.php]: this uses the same "output" from the previous comment

File: inc/html.class.php (hunk 15 above)
Comment (DIOHz0r) [reply inc/html.class.php]: same here

File: inc/html.class.php (hunk 16 above)
Comment (DIOHz0r) [reply inc/html.class.php]: same here

File: inc/html.class.php (hunk 17 above)
Comment (DIOHz0r) [reply inc/html.class.php]: same here

File: inc/html.class.php (hunk 20 above)
Comment (DIOHz0r) [reply inc/html.class.php]: same here

File: inc/html.class.php (hunk 21 above)
Comment (DIOHz0r) [reply inc/html.class.php]: you are returning the result of the date and it returns false or string, so this method does the same

File: inc/html.class.php (hunk 27 above)
Comment (DIOHz0r) [reply inc/html.class.php]: Html::entity_decode_deep and Toolbox::convertImageToTag could return different values so you cant predict which is going to be the result, documenting the return as mixed solves its discrepancy.

File: inc/html.class.php (hunk 29 above)
Comment (DIOHz0r) [reply inc/html.class.php]: same question about printing the output

File: inc/html.class.php (hunk 30 above)
Comment (DIOHz0r) [reply inc/html.class.php]: same question about printing the output

File: inc/html.class.php (hunk 33 above)
Comment (DIOHz0r) [reply inc/html.class.php]: which argument do you mean? if you refer about the value the mixed argument could be a valid option

File: inc/toolbox.class.php (hunk 38 above)
Comment (DIOHz0r) [reply inc/toolbox.class.php]: str_replace retuns mixed results

File: inc/toolbox.class.php (hunk 39 above)
Comment (DIOHz0r) [reply inc/toolbox.class.php]: str_replace retuns mixed results

File: inc/toolbox.class.php (hunk 40 above)
Comment (DIOHz0r) [reply inc/toolbox.class.php]: str_replace retuns mixed results

File: inc/toolbox.class.php (hunk 44 above)
Comment (DIOHz0r) [reply inc/toolbox.class.php]: thx

File: inc/toolbox.class.php (hunk 47 above)
Comment (DIOHz0r) [reply inc/toolbox.class.php]: same as above, also Html::entity_decode_deep could return an array or string.

**Comment by DIOHz0r** [comment]
I have updated the code again, also I have a comment about documentation of the return void cases, they are implied by default for documentation according to PHPDoc (https://docs.phpdoc.org/references/phpdoc/tags/return.html)  and some recommendations are to avoid them (https://stackoverflow.com/questions/2061550/phpdoc-return-void-necessary)

File: inc/toolbox.class.php (hunk 46 above)
Comment (DIOHz0r) [reply inc/toolbox.class.php]: "if correctly used" so this could return mixed values

File: inc/html.class.php (hunk 11 above)
Comment (DIOHz0r) [reply inc/html.class.php]: echo returns a string so at the end this function is returning a string, isn't it?

File: inc/html.class.php (hunk 12 above)
Comment (DIOHz0r) [reply inc/html.class.php]: same issue with echoes

**Pull request closed without merging**